src/core_modules/capture-core/components/Pages/Search/SearchPage.component.js

```diff
@@ -70,24 +76,43 @@ const Index = ({
     preselectedProgramId,
     searchStatus,
 }: Props) => {
-    const [selectedSearchScopeId, setSelectedSearchScopeId] = useState(() => preselectedProgramId);
+    const [selectedSearchScopeId, setSearchScopeId] = useState(() => (preselectedProgramId));
+    const [selectedSearchScopeType, setSearchScopeType] = useState(() => (preselectedProgramId ? searchScopes.PROGRAM : null));
+
+    useEffect(() => {
```

Comment:
Now each time there is a new selection on the Locked selector we want to setSearchScopeId and setSearchScopeType